### PR TITLE
StyleGAN training

### PR DESCRIPTION
**Pull request by xdeng7**
Pull request for adding training for StyleGAN

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1446-1 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1446/1/index.html 
Code coverage of this PR:  vs. Master:

File: scripts/gan/stylegan/README.md [hunk 1]
```diff
@@ -2,7 +2,24 @@
...
+python prepare_data.py --out LMDB_PATH --n_worker N_WORKER DATASET_PATH
+```
+
+Second, train the StyleGAN for FFHQ dataset.
+```bash
+python train.py --path LMDB_PATH --sched
+```
+
+*Notes for training*
+1) The original tensorflow implementation can't be 100% converted by MXNet. Two functions are missing, gradient penalty and blur. The lack of gradient penalty can cause mode collapse while training, so it is neccessary to tune the learning rate based on the number of GPUs and apply early stop. The lack of blur function results in the low image quality and this is one of the important reasons that high-resolution images can't be generted via our implementation.
```
Comment (bryanyzhu) [inline scripts/gan/stylegan/README.md:17]: Please add links to gradient penalty and blur function in original Tensorflow implementation, so that people interested can go check it out. 

Besides, generted -> generated, just a typo.

File: scripts/gan/stylegan/README.md [hunk 2]
```diff
@@ -2,7 +2,24 @@
...
+```
+
+Second, train the StyleGAN for FFHQ dataset.
+```bash
+python train.py --path LMDB_PATH --sched
+```
+
+*Notes for training*
+1) The original tensorflow implementation can't be 100% converted by MXNet. Two functions are missing, gradient penalty and blur. The lack of gradient penalty can cause mode collapse while training, so it is neccessary to tune the learning rate based on the number of GPUs and apply early stop. The lack of blur function results in the low image quality and this is one of the important reasons that high-resolution images can't be generted via our implementation. 
+2) This is an unstable version of StyleGAN. We've tested the training by using 8 GPUs and single GPU. Single GPU can be problematic. The following images are generated by a model trained with 8 GPUs.
```
Comment (bryanyzhu) [inline scripts/gan/stylegan/README.md:18]: This is an unstable version of StyleGAN -> The training of StyleGAN is not stable at this moment due to the aforementioned reasons.

File: scripts/gan/stylegan/README.md [hunk 3]
```diff
@@ -2,7 +2,24 @@
...
+
+Second, train the StyleGAN for FFHQ dataset.
+```bash
+python train.py --path LMDB_PATH --sched
+```
+
+*Notes for training*
+1) The original tensorflow implementation can't be 100% converted by MXNet. Two functions are missing, gradient penalty and blur. The lack of gradient penalty can cause mode collapse while training, so it is neccessary to tune the learning rate based on the number of GPUs and apply early stop. The lack of blur function results in the low image quality and this is one of the important reasons that high-resolution images can't be generted via our implementation. 
+2) This is an unstable version of StyleGAN. We've tested the training by using 8 GPUs and single GPU. Single GPU can be problematic. The following images are generated by a model trained with 8 GPUs. 
+3) It takes around 4 days with 8 GPUs to train a StyleGAN to generate 128x128 images.
```
Comment (bryanyzhu) [inline scripts/gan/stylegan/README.md:19]: What GPU? 8 GPUs -> 8 K80 GPUs

**Review by bryanyzhu — APPROVED** [review]
Minor comments, otherwise LGTM. Thank you for the contribution.

File: scripts/gan/stylegan/README.md (hunk 3 above)
Comment (xdeng7) [reply scripts/gan/stylegan/README.md]: All minor changes have been done. Please check.

**Comment by mli** [comment]
Job PR-1446-2 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1446/2/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1446-3 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1446/3/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-1446-4 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1446/4/index.html 
Code coverage of this PR:  vs. Master:
